mso/resource_mso_site_template_contract_service_graph.go

```diff
@@ -0,0 +1,584 @@
+		apiSiteID := models.StripQuotes(siteCont.S("siteId").String())
+
+		if apiSiteTemplateName == templateName && apiSiteID == siteID {
+			serviceGraphsCount, err := siteCont.ArrayCount("serviceGraphs")
+			if err != nil {
+				return nil, -1, fmt.Errorf("Unable to load site service graphs")
+			}
+
+			for j := 0; j < serviceGraphsCount; j++ {
+				serviceGraphCont, err := siteCont.ArrayElement(j, "serviceGraphs")
+
+				if err != nil {
+					return nil, -1, fmt.Errorf("Unable to load site service graph element")
+				}
+
+				serviceGraphRef := models.StripQuotes(serviceGraphCont.S("serviceGraphRef").String())
+				serviceGraphRefToken := strings.Split(serviceGraphRef, "/")
+
+				if len(serviceGraphRefToken) != 7 {
+					return nil, -1, fmt.Errorf("Invalid site service graph")
+				}
+
+				if schemaID == serviceGraphRefToken[2] && templateName == serviceGraphRefToken[4] && serviceGraphName == serviceGraphRefToken[6] {
+					return serviceGraphCont, j, nil
+				}
+
+			}
+		}
+	}
+	return nil, -1, fmt.Errorf("Unable to find site service graph")
+}
+
+// postSiteTemplateContractServiceGraphConfig create/update a service graph configuration for a site template contract.
+//
+// Parameters:
+//   - ops: The ops to perform create(add)/update(replace) operations.
+//   - d: The schema resource data.
+//   - m: The client interface.
+//
+// Returns:
+//   - An error if there was a problem creating the service graph configuration.
+func postSiteTemplateContractServiceGraphConfig(ops string, d *schema.ResourceData, m interface{}) error {
+	msoClient := m.(*client.Client)
+	schemaID := d.Get("schema_id").(string)
+	templateName := d.Get("template_name").(string)
+	contractName := d.Get("contract_name").(string)
+	siteID := d.Get("site_id").(string)
+
+	var serviceGraphSiteID string
+	if tempServiceGraphSiteID, ok := d.GetOk("service_graph_site_id"); ok {
+		serviceGraphSiteID = tempServiceGraphSiteID.(string)
+	} else {
+		serviceGraphSiteID = siteID
+	}
+
+	serviceGraphRef := make(map[string]interface{})
+	serviceGraphName := d.Get("service_graph_name").(string)
+	serviceGraphRef["serviceGraphName"] = serviceGraphName
+
+	if serviceGraphSchemaId, ok := d.GetOk("service_graph_schema_id"); ok {
+		serviceGraphRef["schemaId"] = serviceGraphSchemaId.(string)
+	} else {
+		serviceGraphRef["schemaId"] = schemaID
+	}
+
+	if serviceGraphTemplateName, ok := d.GetOk("service_graph_template_name"); ok {
+		serviceGraphRef["templateName"] = serviceGraphTemplateName.(string)
+	} else {
+		serviceGraphRef["templateName"] = templateName
+	}
+
+	cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", serviceGraphRef["schemaId"].(string)))
+	if err != nil {
+		return err
+	}
+
+	schemaTemplateServiceGraphCont, _, err := getSchemaTemplateServiceGraph(cont, serviceGraphRef["templateName"].(string), serviceGraphName)
+	if err != nil {
+		return err
+	}
+	apiNodeRelationshipList, err := extractNodes(schemaTemplateServiceGraphCont)
```

Comment:
Need to remove the variable err when the https://github.com/CiscoDevNet/terraform-provider-mso/pull/244 PR is merged with the master.